frontend/src/components/ProgressiveDeliveryComponent/ProgressiveDeliveryComponent.tsx

```diff
@@ -166,19 +165,11 @@ function CustomNodeRenderer({ node: { id } }: DependencyGraphTypes.RenderNodePro
   const paddedWidth = width + padding * 2;
   const paddedHeight = height + padding * 2;
 
-  let node: Node;
-  try {
-    node = JSON.parse(id);
-  } catch {
-    console.warn("Parse error: ", id);
-    console.warn("Assuming this is soak...");
-
+  if (id.startswith(MANY_TO_MANY_NODE_LABEL)) {
```

Comment:
We need to make sure we NEVER have an app called soak ;)

Reply:
good catch! how about 

```suggestion
  if (id.match(new RegExp(`^${L}-\\d+$`))) {
```

Reply:
I committed it, it's better, but on second thought, if the app is named `soak` the `id` will start with `{` as it will be a JSON :)